src/astro/databases/snowflake.py

```diff
@@ -24,6 +25,31 @@ def hook(self):
         """Retrieve Airflow hook to interface with the snowflake database."""
         return SnowflakeHook(snowflake_conn_id=self.conn_id)
 
+    def get_table_qualified_name(self, table: Table):
+        """
+        Return table qualified name for Snowflake.
+
+        :param table: The table we want to retrieve the qualified name for.
+        """
+        if table.metadata is not None and table.metadata.schema is not None:
+            qualified_name = f"{table.metadata.schema}.{table.name}"
+        else:
+            qualified_name = table.name
+        return qualified_name
```

Comment:
Hmm do you think we can move this to BaseDB since it is same for most of at least 3 of them?

Reply:
That was the idea, @kaxil, but I can rollback! What are your thoughts?

Reply:
Yeah I would be in favor of moving this logic to `BaseDatabase.get_table_qualified_name`